src/main.rs

```diff
@@ -77,6 +78,28 @@ impl Device {
     }
 }
 
+fn try_baud_rate(port_name: &str, baud_rate: u32) -> Result<usize, Box<dyn std::error::Error>> {
+    let settings = serialport::SerialPortSettings {
+        baud_rate,
+        data_bits: serialport::DataBits::Eight,
+        flow_control: serialport::FlowControl::None,
+        parity: serialport::Parity::None,
+        stop_bits: serialport::StopBits::One,
+        timeout: std::time::Duration::from_millis(10000),
+    };
+
+    let mut max_byte_read = 0;
+    for _ in 0..NUM_TRIES {
+        let mut serial_port = serialport::open_with_settings(port_name, &settings)?;
+        let mut buffer = vec![0u8; 128];
+        let byte_read = serial_port.read(&mut buffer)?;
+        if byte_read > max_byte_read {
+            max_byte_read = byte_read;
+        }
+    }
+    Ok(max_byte_read)
+}
+
 fn main() {
     let matches = App::new("ublox CLI example program")
         .about("Demonstrates usage of the Rust ublox API")
```

Comment:
Update for non-sample code purpose of the tool now, something like the following.

name: Maivin u-blox configuration tool
about: Configure and test the gps u-blox device